### PR TITLE
Story 16

### PR DESCRIPTION
**Pull request by limsammy**
Related Issue: #12 

Summary: Wrote require_admin method and added a before callback to render 404 for admin paths for user role = default

Author(s): sam

### CODE REVIEW

File: app/views/sessions/new.html.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div>
+<div class="login-wrapper">
```
Comment (randy-darsh) [inline app/views/sessions/new.html.erb:1]: good call

File: app/controllers/admin/admin_controller.rb [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+  before_action :require_admin
 
   def show
     
   end
+
+  private
+
+  def require_admin
+    render file: "/public/404" unless current_admin?
```
Comment (randy-darsh) [inline app/controllers/admin/admin_controller.rb:11]: this is a good idea

**Review by JoelLindow — COMMENTED** [review]
Nice use of the before action. I'm really impressed that you figured this out.

File: app/controllers/admin/base_controller.rb [hunk 3]
```diff
@@ -1,3 +1,7 @@
 class Admin::BaseController < ApplicationController
-  
+  before_action :require_admin
```
Comment (JoelLindow) [inline app/controllers/admin/base_controller.rb:2]: I think it's really good that we check for stuff like this when people log in!

File: spec/features/user/user_is_restricted_from_admin_and_private_data_spec.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+describe "User is restricted from admin pages and other user's data" do
+  context 'User is restricted from other users info' do
+    scenario 'User visits /orders and cannot see other users orders' do
+      user_1 = create(:user, username: 'sam', password: '123')
```
Comment (JoelLindow) [inline spec/features/user/user_is_restricted_from_admin_and_private_data_spec.rb:6]: Thanks for including Sam.

File: spec/features/user/user_is_restricted_from_admin_and_private_data_spec.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      user_2 = create(:user, username: 'joel')
+
+      visit login_path
+
+      fill_in "Username", with: "sam"
+      fill_in "Password", with: "123"
```
Comment (JoelLindow) [inline spec/features/user/user_is_restricted_from_admin_and_private_data_spec.rb:12]: This is a bad password example.

File: spec/features/user/user_is_restricted_from_admin_and_private_data_spec.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+describe "User is restricted from admin pages and other user's data" do
+  context 'User is restricted from other users info' do
+    scenario 'User visits /orders and cannot see other users orders' do
+      user_1 = create(:user, username: 'sam', password: '123')
+      user_2 = create(:user, username: 'joel')
```
Comment (randy-darsh) [inline spec/features/user/user_is_restricted_from_admin_and_private_data_spec.rb:7]: i like that joel is in here

File: spec/features/user/user_is_restricted_from_admin_and_private_data_spec.rb [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  context 'User cannot view admin pages' do
+    scenario 'User visits admin dashboard' do
+      user_1 = create(:user, username: 'sam', password: '123')
+      user_2 = create(:user, username: 'joel')
+
+      visit admin_dashboard_path
+
+      expect(current_path).to eq('/admin/dashboard')
```
Comment (JoelLindow) [inline spec/features/user/user_is_restricted_from_admin_and_private_data_spec.rb:35]: What path is this supposed to be?

**Comment by randy-darsh** [comment]
looks good. merging to dev branch

**Pull request merged**